caravel/assets/javascripts/explorev2/actions/exploreActions.js

```diff
@@ -48,18 +48,14 @@ export function setFilterColumnOpts(filterColumnOpts) {
 }
 
 export function resetFormData() {
-  // Clear all form data when switching datasource
+  // Clear all form data when switching datasource or viz_type
   return { type: RESET_FORM_DATA };
```

Comment:
right..yeah it makes sense to keep form_data used for visualization unchanged (such as show_bar_value) and only reset form_data such as columns/groupby. So I guess this is only needed for switching viz_type then? for instance when we switch from dist_bar to table, form_data such as show_control will be reset to default value false.